process_duty_email.py

```diff
@@ -0,0 +1,293 @@
+        try:
+            process = subprocess.check_output(
+                    [
+                        "powershell",
+                        #"getDNAnexusURL",
+                        "S:\\Genetics_Data2\\Array\\Software\\duty_bioinformatician_scripts\\get_DNAnexus_url.ps1",
+                        urls,
+                    ], 
+                    shell=True,
+                    stderr=subprocess.STDOUT
+                ) 
+            full_process += str(process.decode('utf-8') + "\n")
+        except subprocess.CalledProcessError as e:
+            raise RuntimeError("command '{}' return with error (code {}): {}".format(e.cmd, e.returncode, e.output))
+    return full_process
+
+def get_data_GSTT(df, path_to_folder):
+    """
+    This function reads the data frame (df)..
+    From the df file it obtains the download links for files to be used by Viapath at GSTT creates a string argument.
+    
+    """
+    GSTT_urls = ""
+    for index, row in df.iterrows():
+        if not bool(re.search("StG", row["url"])):  
+            line = row['url'] + "," + path_to_folder + "£$%"
+            GSTT_urls += line
+    return GSTT_urls
+
+def get_data_StG(df, path_to_folder):
+    """
+    This function reads the data frame (df)..
+    From the df file it obtains the download links for files destined for StG transfer and creates a string argument.
+    
+    """
+    StG_urls = ""
+    for index, row in df.iterrows():
+        if bool(re.search("StG", row["url"])):  
+            line = row['url'] + "," + path_to_folder + "£$%"
+            StG_urls += line
+    return StG_urls
+
+def save_log_file(text, filename): 
+    cur_time = datetime.now()
+    cur_time_string = cur_time.strftime("%Y_%m_%d__%H_%M_%S_")
+    log_path = "{}\\process_logs\\Finished_on_{}_{}.txt".format(config.CSVread_folder, cur_time_string, filename)
+    with open(log_path, 'w') as f:
+        f.write(text)
+
+def get_files():
+    """
+    This function gets the paths for csv files download by duty bioinformatician to the Duty_Bioinformatics_CSV folder.
+    """
+    folder = config.CSVread_folder
+    files = (
+        file
+        for file in os.listdir(folder)
+        if os.path.isfile(os.path.join(folder, file))
+    )
+    files_list = collections.defaultdict(list)
+    for filename in files:
+        results = re.search("__(\S+)__", filename)
+        project = results.group(1)
+        filepath = folder + "\\" + filename
+        files_list[project].append(filepath)
+    files_merged = dict(files_list)
+    return files_merged
+
+class Project:
+    """
+    Process project
+    """
+    def __init__(self, project, files):
+        self.project = project
+        self.project_files = files
+    def process_data(self):
+        """
+        This function executes the processing of a project based on the project type e.g. SNP, WES, TSO500 or MokaPipe
+        """
+        if self.project == "WES":
+            Project.process_WES(self)

```

Comment:
I think this should be self.process_WES()

Reply:
changed it to self.